src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -165,7 +165,7 @@ public function getDecoratedService()
     /**
      * Sets the service class.
      *
-     * @param string $class The service class
+     * @param string|Parameter|null $class The service class
```

Comment:
That one gave me quite a headache. I will open a PR on 3.4 to adjust the type-hint.